src/TBuffer.cpp

```diff
@@ -5093,3 +5094,20 @@ QString TBuffer::processSupportsRequest(const QString& elements)
 
     return result.join(QLatin1String(" "));
 }
+
+// Count the graphemes in a QString - returning it's length in terms of those:
+unsigned int TBuffer::graphemeLength(const QString& text)
```

Comment:
:thinking: Humm, in this case I guess it could be okay to have ~`singed`~  `signed int`s.

However in other places in the `TBuffer` class it is a real PITA that the formatting attributes are in a `std::deque<std::deque<TChar>>` type which use the **`unsigned`** `std::size_t` (directly from the C `size_t` type) for indexing which is by definition always the right size to use for indexing into arrays whereas the textual content is based upon `QList<QList<QChar>>` or rather `QStringList` which uses `int`s for indexing. Making the code seem type safe (and getting rid of all the clang code warnings about implicit casts between signed and unsigned integer types) with explicit `static_cast<Ta>(value of type Tb)` is not my idea of fun...

Reply:
In next commit.

As it happens, with the removal of the code to add words directly from the game server's output this method is not currently used.  I expect however to need it in the very near future (along with a `graphemeMid(int start, int length = -1)`) to better handle the text selection/highlighting issues that we have had for a long time now and which I promised to get onto, was it, *last* week. :worried: